src/controller/java/AndroidCallbacks.cpp

```diff
@@ -78,6 +79,11 @@ void GetConnectedDeviceCallback::OnDeviceConnectedFn(void * context, Operational
     VerifyOrReturn(successMethod != nullptr, ChipLogError(Controller, "Could not find onDeviceConnected method"));
 
     static_assert(sizeof(jlong) >= sizeof(void *), "Need to store a pointer in a Java handle");
+
+    // TODO (#) This is a memory leak since as of today this is never freed. Either we need to make sure this is free
```

Comment:
This is a pretty big TODO that I am not sure how to do, I am seeking recommendation on possible suggestions on how to eliminate this memory leak since I have wasted over 1 full working day on this